backend/game/src/gameplay/game_event_handler.rs

```diff
@@ -507,18 +507,38 @@ pub fn update_game_state(
             let Some(state) = &mut game_state.agenda else {
                 bail!("agenda state not initialized, this is a bug.");
             };
-            let Some(_vote) = &state.vote else {
+            let Some(vote) = &state.vote else {
                 bail!("no agenda has been revealed yet");
             };
+            if let Some(outcome) = &outcome {
+                ensure!(
+                    AgendaElectKind::from(outcome) == vote.elect,
+                    "invalid elect kind, expected {:?}",
+                    vote.elect
+                )
+            }
+
+            let vote = state.vote.take().unwrap();
             state.round = state.round.next();
 
-            // TODO: log the outcome of the vote
+            let Some(outcome) = outcome else { todo!() };
```

Comment:
Är `NONE` veto?

Reply:
right forgot about this todo

Reply:
None är inte veto btw, None är bara "discard without effect", i.e. 'Deadly Plot' action card

Reply:
Right